frontend/packages/ux-editor/src/components/config/editModal/EditBooleanValue.tsx

```diff
@@ -8,32 +8,37 @@ import { useComponentPropertyLabel } from '../../../hooks/useComponentPropertyLa
 export interface EditBooleanValueProps extends IGenericEditComponent {
   propertyKey: string;
   helpText?: string;
+  defaultValue?: boolean;
 }
 
 export const EditBooleanValue = ({
   component,
   handleComponentChange,
   propertyKey,
   helpText,
+  defaultValue,
 }: EditBooleanValueProps) => {
   const t = useText();
   const componentPropertyLabel = useComponentPropertyLabel();
 
   const handleChange = () => {
     handleComponentChange({
       ...component,
-      [propertyKey]: !component[propertyKey],
+      [propertyKey]: getNewBooleanValue(),
     });
   };
 
   const isValueExpression = (value: any) => {
     return Array.isArray(value);
   };
 
+  const getNewBooleanValue = () =>
+    component[propertyKey] === undefined ? !defaultValue : !component[propertyKey];
```

Comment:
Same as prev comment. 😊 
```suggestion
 const getNewBooleanValue = () => !(component[propertyKey] ?? defaultValue);
```